packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/web-services/socket-session-endpoint.ts

```diff
@@ -0,0 +1,203 @@
+
+    const requestBody = this.createRequestBody(baseConfig, methodName, args);
+    const response = await this.transaction.transact(requestBody);
+    const str_response = String(response.transactionReceipt);
+
+    if (str_response.includes("Query response error")) {
+      if (str_response.includes(`"errorCode":3`)) {
+        this.log.debug("Waiting for new blocks...");
+      } else {
+        this.log.error(`Runtime error caught: ${str_response}`);
+      }
+    } else {
+      this.log.debug(`New block found`);
+      transactionReceipt = response.transactionReceipt;
+      next = { transactionReceipt };
+      this.socket.emit(IrohaSocketSession.Next, next);
+      this.currentBlockHeight++;
+    }
+  }
+
+  public async startMonitor(monitorOptions: any): Promise<void> {
+    this.log.debug(`${IrohaSocketSession.Subscribe} => ${this.socket.id}`);
+    this.log.info(`Starting monitoring blocks...`);
+
+    this.monitorMode = true;
+    await this.getInitialMaxBlockHeight(monitorOptions.baseConfig);
+
+    this.monitoringInterval = setInterval(() => {
+      this.monitoringRoutine(monitorOptions.baseConfig);
+    }, monitorOptions.pollTime);
+  }
+
+  private async validateMethodName(methodName: string): Promise<boolean> {
+    let isValidMethod = false;
+
+    if (Object.values(IrohaQuery as any).includes(methodName)) {
+      this.log.debug(`Method name: ${methodName} (IrohaQuery) is valid`);
+      isValidMethod = true;
+    }
+    if (Object.values(IrohaCommand as any).includes(methodName)) {
+      this.log.debug(`Method name: ${methodName} (IrohaCommand) is valid`);
+      isValidMethod = true;
+    }
+    return isValidMethod;
+  }
+
+  public async stopMonitor(): Promise<void> {
+    this.log.info(`Stopping monitor...`);
+    this.monitorMode = false;
+    clearInterval(this.monitoringInterval);
+  }
+
+  public async sendAsyncRequest(asyncRequestData: any): Promise<any> {
+    this.log.debug(`Inside ##sendAsyncRequest()`);
+    this.log.debug(
+      `asyncRequestData: ${JSON.stringify(asyncRequestData, null, 4)}`,
+    );
+
+    this.log.debug(asyncRequestData.methodName);
+    this.log.debug(asyncRequestData.args);
+    this.log.debug(asyncRequestData.baseConfig);
+
+    // Validate method name
+    if ((await this.validateMethodName(asyncRequestData.methodName)) === true) {
+      const requestBody = this.createRequestBody(
+        asyncRequestData.baseConfig,
+        asyncRequestData.methodName,
+        asyncRequestData.args,
+      );
+
+      const response = await this.transaction.transact(requestBody);
+      if (asyncRequestData?.syncRequest) {
+        this.log.debug(response);
+        return response;
+      }
+    } else {
+      const error = `Unrecognized method name: ${asyncRequestData.methodName}`;
+      this.log.debug(error);
+    }
+  }
+
+  public async sendSyncRequest(syncRequestData: any): Promise<void> {
+    this.log.debug(`Inside ##sendSyncRequest()`);
+
+    syncRequestData["syncRequest"] = true;
```

Comment:
@StepniowskiP Could this input mutation be avoided by having the boolean flag as a separate parameter of the `sendAsyncRequest` method?

Reply:
I have added flag parameter async to sendRequest function